service/src/test/java/com.codeforcommunity/processor/ReservationProcessorImplTest.java

```diff
@@ -0,0 +1,330 @@
+import com.codeforcommunity.enums.ReservationAction;
+import com.codeforcommunity.exceptions.AuthException;
+import com.codeforcommunity.exceptions.IncorrectBlockStatusException;
+import java.util.Optional;
+import org.jooq.generated.tables.records.ReservationsRecord;
+import org.junit.jupiter.api.Test;
+
+public class ReservationProcessorImplTest {
+  JooqMock mockDb;
+  ReservationProcessorImpl proc;
+
+  /**
+   * Method to setup mockDb and processor
+   *
+   * <p>TODO: This can't be run with the @BeforeEach annotation by maven currently. The way to
+   * enable this is outlined in this Stack Overflow post:
+   * https://stackoverflow.com/questions/51382356/maven-does-not-run-beforeeach-methods-while-running
+   * This isn't currently being done because our version of maven surefire is auto imported and is
+   * only version 2.12.4
+   */
+  void setup() {
+    mockDb = new JooqMock();
+    proc = spy(new ReservationProcessorImpl(mockDb.getContext()));
+  }
+
+  @Test
+  public void testIsAdmin() {
+    setup();
+    for (PrivilegeLevel p : PrivilegeLevel.values()) {
+      boolean isAdmin = p.equals(PrivilegeLevel.ADMIN) || p.equals(PrivilegeLevel.SUPER_ADMIN);
+      try {
+        this.proc.isAdminCheck(p);
+        if (!isAdmin) {
+          fail("Method should've thrown an AuthException with privilege level " + p.getName());
+        }
+      } catch (AuthException e) {
+        if (isAdmin) {
+          fail(
+              "Method should not have thrown an AuthException with privilege level " + p.getName());
+        }
+      }
+    }
+
+    try {
+      this.proc.isAdminCheck(PrivilegeLevel.ADMIN);
+    } catch (AuthException e) {
+      fail("Method should not have thrown an exception");
+    }
+
+    try {
+      this.proc.isAdminCheck(PrivilegeLevel.SUPER_ADMIN);
+    } catch (AuthException e) {
+      fail("Method should not have thrown an exception");
+    }
+  }
+
+  @Test
+  public void testBlockOpenCheckNoAction() {
+    setup();
+    doReturn(Optional.empty()).when(this.proc).lastAction(anyInt());
+
+    try {
+      this.proc.blockOpenCheck(1);
+    } catch (IncorrectBlockStatusException e) {
+      fail("Method should not throw exception");
+    }
+  }
+
+  @Test
+  public void testBlockOpenCheckUncomplete() {
+    setup();
+    ReservationsRecord mockReservation = mock(ReservationsRecord.class);
+    doReturn(Optional.of(mockReservation)).when(this.proc).lastAction(anyInt());
+
+    when(mockReservation.getActionType())
+        .thenReturn(ReservationAction.UNCOMPLETE, ReservationAction.UNCOMPLETE);
+
+    try {
+      this.proc.blockOpenCheck(1);
+    } catch (IncorrectBlockStatusException e) {
+      fail("Method should not throw exception");
+    }
+  }
+
+  @Test
+  public void testBlockOpenCheckReserved() {
+    setup();
+    ReservationsRecord mockReservation = mock(ReservationsRecord.class);
+    doReturn(Optional.of(mockReservation)).when(this.proc).lastAction(anyInt());
```

Comment:
Similarly to above, in the case where you need to return a valid value, you can call `mockDb.addReturn("SELECT", mockReservation)` (or `mockDb.addReturn(OperationType.SELECT, mockReservation)` after the new `JooqMock` PR gets merged)